.github/workflows/build-and-deploy-site.yml

```diff
@@ -0,0 +1,53 @@
+name: Build and Deploy Site
+
+on:
+  push:
+    branches: [ "main", "html_page" ]
```

Comment:
(1) Are you sure you need to build-and-deploy on every push to master?
(2) Are you sure the branch `html_page` will exist after the PR from the `html_page` branch will be merged into `master`?

Reply:
Also, seems like will need an exception for "gh-pages" branch if you choose to trigger on "all" pushes/PRs.